spatialprofilingtoolbox/cggnn/extract.py

```diff
@@ -0,0 +1,72 @@
+"""Extract information cg-gnn needs from SPT."""
+
+from pandas import DataFrame, concat, merge  # type: ignore
+from numpy import sort  # type: ignore
+
+from spatialprofilingtoolbox.db.feature_matrix_extractor import FeatureMatrixExtractor
+
+
+def _create_cell_df(dfs_by_specimen: dict[str, DataFrame]) -> DataFrame:
+    """Find simple and complex phenotypes, and locations and merge into a DataFrame."""
+    for specimen, df_specimen in dfs_by_specimen.items():
+        df_specimen['specimen'] = specimen
+
+    df = concat(dfs_by_specimen.values(), axis=0)
+    df.index.name = 'histological_structure'
+    # Reorder columns so it's specimen, xy, channels, and phenotypes
+    column_order = ['specimen', 'pixel x', 'pixel y']
+    column_order.extend(df.columns[df.columns.str.startswith('C ')])
+    column_order.extend(df.columns[df.columns.str.startswith('P ')])
+    return df[column_order]
+
+
+def _create_label_df(
+    df_assignments: DataFrame,
+    df_strata: DataFrame,
+    strata_to_use: list[int] | None,
+) -> tuple[DataFrame, dict[int, str]]:
+    """Get slide-level results."""
+    df_assignments = df_assignments.set_index('specimen')
+    df_strata = df_strata.set_index('stratum identifier')
+
+    # Filter for strata to use
```

Comment:
I would prefer that we use functions `filter_for_strata_to_use`, `drop_same_columns`, `compress_remaining_columns_into_str`, etc., instead of comment-headed blocks. This would break down the function into smaller units but also reduce machine-unaccountable comments.
This is not essential however and shouldn't block merge.